src/api.jl

```diff
@@ -12,26 +12,41 @@ struct SurrogateGenerator{S<:Surrogate, X, Y, A, R<:AbstractRNG}
 end
 
 """
-    surrogenerator(x, method::Surrogate [, rng]) → sg::SurrogateGenerator
+    surrogenerator(x, method::Surrogate [, rng]) → sgen::SurrogateGenerator
 
 Initialize a generator that creates surrogates of `x` on demand, based on the given `method`.
-This is efficient, because for most methods some things can be initialized and reused
+This is more efficient than [`surrogate`](@ref),
+because for most methods some things can be initialized and reused
 for every surrogate. Optionally you can provide an `rng::AbstractRNG` object that will
 control the random number generation and hence establish reproducibility of the
 generated surrogates. By default `Random.default_rng()` is used.
 
-Notice that the generated surrogates overwrite, in-place, a common vector container.
-Use copy if you need to actually store multiple surrogates.
+Tthe generated surrogates overwrite, in-place, a common vector container.
+Use `copy` if you need to actually store multiple surrogates.
 
-To generate a surrogate, call `sg` as a function with no arguments, e.g.:
+To generate a surrogate, call `sgen` as a function with no arguments, e.g.:
 
 ```julia
-sg = surrogenerator(x, method)
-for i in 1:1000
-    s = sg()
-    # do stuff with s and or x
-    result[i] = stuff
-end
+sgen = surrogenerator(x, method)
+s = sgen()
+```
+
+Notice that you can use the generator syntax of Julia to map over surrogates
+generated by `sg`. For example, let `q` be a function returning a discriminatory statistic.
+In the typical application scenario of TimeseriesSurrogates.jl you'd do
+
+```julia
+using TimeseriesSurrogates
+q, x # inputs
+method = RandomFourier() # some example method
+sgen = surrogenerator(x, method)
+siter = (sgen() for _ in 1:1000)
+qx = q(x)
+qs = map(q, siter)
+# compare `qx` with quantiles
+using Statistics: quantile
+q01, q99 = quantile(qs, [0.01, 0.99])

```

Comment:
It's up to the researcher to decide the bounds here, but I've never seen a 98% range used before. Perhaps change to `qlow, qhigh = quantile(qs, [0.005, 0.995])`?

Reply:
Yes, I know it is uncommon. I have followed some very very good lectures on statistics and causal inference stuff from Richard McElreath https://www.youtube.com/watch?v=cclUd_HoRlo&list=PLDcUM9US4XdMROZ57-OIRtIK0aOynbgZN&index=2  truly transformative. Anyways, one of the things he does in these courses intentially is to use somewhat uncommon confidence intervals exactly to get the point across that there is nothing magical about 5-95 CI or any other. I got used to it and now I kinda do it myself also. If you don't mind the 01-99 I'd like ti keep it :D

Reply:
> Yes, I know it is uncommon. I have followed some very very good lectures on statistics and causal inference stuff from Richard McElreath https://www.youtube.com/watch?v=cclUd_HoRlo&list=PLDcUM9US4XdMROZ57-OIRtIK0aOynbgZN&index=2 truly transformative. Anyways, one of the things he does in these courses intentially is to use somewhat uncommon confidence intervals exactly to get the point across that there is nothing magical about 5-95 CI or any other. I got used to it and now I kinda do it myself also. If you don't mind the 01-99 I'd like ti keep it :D

That's actually a really good argument. Let's keep it as is. 

I do something similar to nail the point that there's nothing special about variable names are when teaching introductory programming. For example: "There's nothing special about the variable name `x`, let's call it `chicken_nuggets` instead and see what happens".